### PR TITLE
Core Contract Cache improvements and bugfixes

### LINKED ISSUE #602: Investigate why CeloVoterVotes fetcher is not processing items on rc1staging
As discussed yesterday, there is an issue with the Celo Voter Votes fetcher not correctly processing items from the database. This needs to be investigated and resolved as some significant functionality relies on this working correctly.

### PR DESCRIPTION
**Pull request by rkachowski**
### Description

Related to the inability to fetch new epoch data as described in #602 is the problem of refreshing the contract cache. The synchronous nature of the refresh request processing in the currently live instance prevents introspection, but the same issue can be reproduced artificially with unit tests by having failed refresh calls. This in turn prevents contracts from being recognised as core contracts and everything will return negative.

> tl;dr - the address refresh was both slow and broken, but should now be fast and not broken.
 
 ### Other changes

* Refreshing addresses with async tasks, so other processes are in charge of making the actual blockchain requests
    * This means that the cache is unblocked to serve address requests (except in the case of a cache miss, where the only response is to block until the answer is retrieved and stored)
* Added more comments and logging 

### Tested

* Created and ran unit tests

### Issues

 - Fixes #602
 
 ### Backwards compatibility

They're backwards compatible

### CODE REVIEW

File: apps/explorer/test/explorer/celo/core_contract_cache_test.exs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        {CoreContracts,
+         %{
+           refresh_period: :timer.hours(1000),
+           cache: %{test_contract_identifier => test_address}
+         }}
+      )
+
+      assert CoreContracts.is_core_contract_address?(test_address)
+
+      test_address = "0xnewaddresshey"
```
Comment (obasilakis) [inline apps/explorer/test/explorer/celo/core_contract_cache_test.exs:24]: is this line necessary?

**Review by obasilakis — APPROVED** [review]

File: apps/explorer/test/explorer/celo/core_contract_cache_test.exs (hunk 1 above)
Comment (rkachowski) [reply apps/explorer/test/explorer/celo/core_contract_cache_test.exs:24]: no it shouldn't be there, nice catch

**Review by SimsonEnduro — APPROVED** [review]

**Review by SimsonEnduro — APPROVED** [review]
13c137570b90df75324f9cd259a75e50e3e65bb5

**Review by SimsonEnduro — APPROVED** [review]

**Comment by SimsonEnduro** [comment]
a1df26f0427ed23523f887a4e29bc157c51fb0e9
